allensdk/brain_observatory/behavior/behavior_session.py

```diff
@@ -1174,33 +1187,164 @@ def metadata(self) -> Dict[str, Any]:
         return self._get_metadata(behavior_metadata=self._metadata)
 
     @classmethod
-    def _read_data_from_stimulus_file(
-            cls, stimulus_file: BehaviorStimulusFile,
+    def _read_licks(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
+            sync_file: Optional[SyncFile]) -> Licks:
+        """
+        Construct the Licks data object for this session
+        """
+
+        stimulus_timestamps = cls._read_behavior_stimulus_timestamps(
+                        sync_file=sync_file,
+                        stimulus_file_lookup=stimulus_file_lookup,
+                        monitor_delay=0.0)
+
+        return Licks.from_stimulus_file(
+            stimulus_file=stimulus_file_lookup.behavior_stimulus_file,
+            stimulus_timestamps=stimulus_timestamps)
+
+    @classmethod
+    def _read_rewards(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
+            sync_file: Optional[SyncFile]) -> Rewards:
+        """
+        Construct the Rewards data object for this session
+        """
+        stimulus_timestamps = cls._read_behavior_stimulus_timestamps(
+                        sync_file=sync_file,
+                        stimulus_file_lookup=stimulus_file_lookup,
+                        monitor_delay=0.0)
+
+        return Rewards.from_stimulus_file(
+            stimulus_file=stimulus_file_lookup.behavior_stimulus_file,
+            stimulus_timestamps=stimulus_timestamps.subtract_monitor_delay())
+
+    @classmethod
+    def _read_stimuli(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
             stimulus_timestamps: StimulusTimestamps,
+            stimulus_presentation_columns: Optional[List[str]] = None
+    ) -> Stimuli:
+        """
+        Construct the Stimuli data object for this session
+        """
+        return Stimuli.from_stimulus_file(
+                    stimulus_file=stimulus_file_lookup.behavior_stimulus_file,
+                    stimulus_timestamps=stimulus_timestamps,
+                    presentation_columns=stimulus_presentation_columns)
+
+    @classmethod
+    def _read_trials(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
+            sync_file: Optional[SyncFile],
+            monitor_delay: float,
+            licks=Licks,
```

Comment:
I think you meant for this to be a type, not the default value (same for rewards)